src/spans/awsSpan.ts

```diff
@@ -346,11 +347,16 @@ export const getHttpSpan = (
     HTTP_SPAN
   );
 
-  const info = Object.assign({}, basicHttpSpan.info, {
+  let info = Object.assign({}, basicHttpSpan.info, {
     httpInfo,
     ...awsServiceData,
   });
 
+  // add messageId based on W3cContextPropagation in case of messageId not present
+  if (!info.messageId) {
+    info.messageId = getW3CMessageId(requestData.headers);
```

Comment:
I thought about it - but it brings complications since there are objects assignment and overrides of properties. 
I'd instead do it in a more precise way like this.